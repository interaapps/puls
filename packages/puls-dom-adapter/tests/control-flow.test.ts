import {PulsDOMAdapter} from "../index";
import {createTemplateFunction} from "pulsjs-template";

const html = createTemplateFunction()
test('dom-control-flow-if-true', () => {
    const d = new PulsDOMAdapter(html`
        <div :if=${true}>
            Hello World
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-false expects none', () => {
    const d = new PulsDOMAdapter(html`
        <div :if=${false}>
            Hello World
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-1 expects 0', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <div :if=${true}>
                Hello World
            </div>
            <div :else>
                Hello World 2
            </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-2 expects 2', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <div :if=${false}>
                Hello World
            </div>
            <div :else>
                Hello World 2
            </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-if-1 expects 0', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <div :if=${true}>
                Hello World
            </div>
            <div :else-if=${false}>
                Hello World 2
            </div>
            <div :else>
                Hello World 3
            </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-if-2 expects 0', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <div :if=${true}>
                Hello World
            </div>
            <div :else-if=${true}>
                Hello World 2
            </div>
            <div :else>
                Hello World 3
            </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-if-3 expects 2', () => {
    const d = new PulsDOMAdapter(html`
        <div>
        <div :if=${false}>
            Hello World
        </div>
        <div :else-if=${true}>
            Hello World 2
        </div>
        <div :else>
            Hello World 3
        </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-if-4 expects 3', () => {
    const d = new PulsDOMAdapter(html`
        <div>
        <div :if=${false}>
            Hello World
        </div>
        <div :else-if=${false}>
            Hello World 2
        </div>
        <div :else>
            Hello World 3
        </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
test('dom-control-flow-if-else-if-5 expects 2', () => {
    const d = new PulsDOMAdapter(html`
        <div>
        <div :if=${false}>
            Hello World
        </div>
        <div :else-if=${true}>
            Hello World 2
        </div>
        <div :else-if=${true}>
            Hello World 3
        </div>
        <div :else>
            Hello World 4
        </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})

test('dom-control-flow-if-else-if-6 expects 3', () => {
    const d = new PulsDOMAdapter(html`
        <div>
        <div :if=${false}>
            Hello World
        </div>
        <div :else-if=${false}>
            Hello World 2
        </div>
        <div :else-if=${true}>
            Hello World 3
        </div>
        <div :else>
            Hello World 4
        </div>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})
