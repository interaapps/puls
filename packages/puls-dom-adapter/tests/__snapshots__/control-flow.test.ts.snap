// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dom-control-flow-if-else-1 expects 0 1`] = `
<div>
  <div>
    Hello World
            
  </div>
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-else-2 expects 2 1`] = `
<div>
  <!--if-->
  <div>
    Hello World 2
            
  </div>
</div>
`;

exports[`dom-control-flow-if-else-if-1 expects 0 1`] = `
<div>
  <div>
    Hello World
            
  </div>
  <!--if-->
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-else-if-2 expects 0 1`] = `
<div>
  <div>
    Hello World
            
  </div>
  <!--if-->
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-else-if-3 expects 2 1`] = `
<div>
  <!--if-->
  <div>
    Hello World 2
        
  </div>
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-else-if-4 expects 3 1`] = `
<div>
  <!--if-->
  <!--if-->
  <div>
    Hello World 3
        
  </div>
</div>
`;

exports[`dom-control-flow-if-else-if-5 expects 2 1`] = `
<div>
  <!--if-->
  <div>
    Hello World 2
        
  </div>
  <!--if-->
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-else-if-6 expects 3 1`] = `
<div>
  <!--if-->
  <!--if-->
  <div>
    Hello World 3
        
  </div>
  <!--if-->
</div>
`;

exports[`dom-control-flow-if-false expects none 1`] = `<!--if-->`;

exports[`dom-control-flow-if-true 1`] = `
<div>
  Hello World
        
</div>
`;
